rules/custom_types_test.go

```diff
@@ -0,0 +1,206 @@
+package rules
+
+import (
+	"testing"
+
+	"github.com/ghodss/yaml"
+)
+
+func TestRuleGroupUnmarshalJSON(t *testing.T) {
+	for _, testCase := range []struct {
+		name string
+		raw  []byte
+		out  RuleGroup
+		err  bool
+	}{
+		{
+			name: "almost empty",
+			raw:  []byte("{}"),
+		},
+		{
+			name: "one recording rule",
+			raw: []byte(`name: foo
+interval: 5s
+rules:
+- record: bar
+  expr: vector(1)`),
+			out: RuleGroup{
+				Name:     "foo",
+				Interval: "5s",
+				Rules: []interface{}{
+					RecordingRule{
+						Record: "bar",
+						Expr:   "vector(1)",
+						Labels: RecordingRule_Labels{AdditionalProperties: make(map[string]string)},
+					},
+				},
+			},
+		},
+		{
+			name: "one alerting rule",
+			raw: []byte(`name: foo
+interval: 5s
```

Comment:
I think would be more readable if we format/align the content of `raw` (same in the other test inputs)

Reply:
This is a literal YAML string, and indentation matters in YAML, that's why it's like this :/

Reply:
We could instead add an extra newline before the first line so they are all aligned totally to the left. WDYT?

Reply:
ah right :/
cool, I think doing as you suggested reads better!

Reply:
OK, updated :heart: